config/service_providers.yml

```diff
@@ -6,6 +6,8 @@ test:
     block_encryption: 'none'
```

Comment:
Also, not all of the agencies in here have an agency ID. Is that intentional? What do we expect to happen for SPs without an agency ID?

Reply:
sp's that do not have an agency id default to their id as before.

Reply:
Hmm, it seems to me like we'd want to make that a required field, right? It seems like the goal should be to stop sharing things from the identities table except the basic handoff info, using the agency identity table instead. I may be off base, what's the motivation for keeping the UUIDs in the identity table?

Reply:
Eventually I'd like to make it required if we can get all the mappings.  However it will have to be done in two migrations:  add column (allow null), populate data, modify column (not null).  The motivation behind keeping uuids in identities is we can set the new uuids feature flag off (ie rollback) with zero downtime and the system will behave as before.  I mentioned we can eventually get rid of this column after the rollout is deemed a success by all the agencies.  However, now it seems there is added incentive to keep that uuid column in identities with the ability to delete users and prevent a uuid from being reused again since the uuid will definitely be deleted from the agency_identities table (since we want to stop charging the agency for that user).

Reply:
Yeah, I get that we need to keep it for rolling back. I'm not suggesting we drop the column all together. But, if we want to make it required in the future, I vote that we just make it required now and avoid another migration in the future where we have to do the same move we are now for SPs that don't have an associated agency.

Reply:
The uuids are currently required in identities.  If you are talking about making the agency_id required if we can get the rest of the sp's mapped to agencies I'm all for it.  I can initially populate it with some some default value when I add the column and then let the seeder update it appropriately.

Reply:
Yeah, I'm talking about make the agency id required so all of the SPs can move to the agency UUIDs in one move instead of having to come back in do it again later.

Reply:
An interesting thing to look at will also be how the dashboard creates service providers. We'll want to make sure the way we specify agency there is compatible.

Reply:
Okay, Steve and I chatted about the agency ID thing and Slack and both agreed that it is probably better to make it not required, so that we don't have to migrate all agencies at one time. Once all of the SPs have an agency ID, then we will make it required.